secant/Stubs/MockServices.swift

```diff
@@ -65,7 +57,69 @@ class MockMnemonicPhraseHandling: MnemonicSeedPhraseHandling {
     func isValid(mnemonic: String) throws {}
 }
 
+class KeysPresentStub: KeyStoring {
+    init(returnBlock: @escaping () throws -> Bool) {
+        self.returnBlock = returnBlock
+    }
+    var returnBlock: () throws -> Bool
+    var called = false
+    func areKeysPresent() throws -> Bool {
+        called = true
+        return try returnBlock()
+    }
+    
+    var birthday: BlockHeight?
+    var phrase: String?
+    func importBirthday(_ height: BlockHeight) throws {
+        guard birthday == nil else {
+            throw KeyStoringError.alreadyImported
+        }
+        birthday = height
+    }
+    
+    func exportBirthday() throws -> BlockHeight {
+        guard let birthday = birthday else {
+            throw KeyStoringError.uninitializedWallet
+        }
+        return birthday
+    }
+    
+    func importPhrase(bip39 phrase: String) throws {
+        guard self.phrase == nil else {
+            throw KeyStoringError.alreadyImported
+        }
+        self.phrase = phrase
+    }
+    
+    func exportPhrase() throws -> String {
+        guard let phrase = self.phrase else {
+            throw KeyStoringError.uninitializedWallet
+        }
+        return phrase
+    }
+    
+    var keysPresent: Bool {
+        return self.phrase != nil && self.birthday != nil
+    }
+    
+    func nukePhrase() {
+        self.phrase = nil
+    }
+    
+    func nukeBirthday() {
+        self.birthday = nil
+    }
+    
+    func nukeWallet() {
+        nukePhrase()
+        nukeBirthday()
+    }
+}
 class MockKeyStoring: KeyStoring {
+    func areKeysPresent() throws -> Bool {
```

Comment:
This function should come after the properties.  The keysPresent computed property could be moved up too to come just after these properties here.